src/ui/Facet/FacetSearchElement.ts

```diff
@@ -0,0 +1,139 @@
+    this.search = document.createElement('div');
+    $$(this.search).addClass('coveo-facet-search');
+
+    this.magnifier = document.createElement('div');
+    this.magnifier.innerHTML = SVGIcons.icons.search;
+    $$(this.magnifier).addClass('coveo-facet-search-magnifier');
+    SVGDom.addClassToSVGInContainer(this.magnifier, 'coveo-facet-search-magnifier-svg');
+    this.search.appendChild(this.magnifier);
+
+    this.wait = document.createElement('div');
+    this.wait.innerHTML = SVGIcons.icons.loading;
+    $$(this.wait).addClass('coveo-facet-search-wait-animation');
+    SVGDom.addClassToSVGInContainer(this.wait, 'coveo-facet-search-wait-animation-svg');
+    this.search.appendChild(this.wait);
+    this.hideFacetSearchWaitingAnimation();
+
+    this.clear = $$(
+      'div',
+      { className: 'coveo-facet-search-clear', title: l('Clear', l('Search')) },
+      SVGIcons.icons.checkboxHookExclusionMore
+    ).el;
+    SVGDom.addClassToSVGInContainer(this.clear, 'coveo-facet-search-clear-svg');
+    this.clear.style.display = 'none';
+    this.search.appendChild(this.clear);
+
+    const middle = document.createElement('div');
+    $$(middle).addClass('coveo-facet-search-middle');
+    this.search.appendChild(middle);
+
+    this.input = document.createElement('input');
+    this.input.setAttribute('type', 'text');
+    this.input.setAttribute('autocapitalize', 'off');
+    this.input.setAttribute('autocorrect', 'off');
+    $$(this.input).addClass('coveo-facet-search-input');
+    Component.pointElementsToDummyForm(this.input);
+    middle.appendChild(this.input);
+
+    $$(this.input).on('keyup', (e: KeyboardEvent) => {
+      handleFacetSearchKeyUp(e);
+    });
+    $$(this.clear).on('click', (e: Event) => {
+      handleFacetSearchClear();
+    });
+    $$(this.input).on('focus', (e: Event) => {
+      handleFacetSearchFocus();
+    });
+
+    this.detectSearchBarAnimation();
+    return this.search;
+  }
+
+  public showFacetSearchWaitingAnimation() {
+    $$(this.magnifier).hide();
+    $$(this.wait).show();
+  }
+
+  public hideFacetSearchWaitingAnimation() {
+    $$(this.magnifier).show();
+    $$(this.wait).hide();
+  }
+
+  public detectSearchBarAnimation() {
+    EventsUtils.addPrefixedEvent(this.search, 'AnimationStart', event => {
+      if (event.animationName == 'grow') {
+        this.searchBarIsAnimating = true;
+      }
+    });
+
+    EventsUtils.addPrefixedEvent(this.search, 'AnimationEnd', event => {
+      if (event.animationName == 'grow') {
+        this.searchBarIsAnimating = false;
+      }
+    });
+  }
+
+  public positionSearchResults(root: HTMLElement, facetWidth: number, nextTo: HTMLElement) {
+    if (this.searchResults != null) {
+      root.appendChild(this.searchResults);
+      this.searchResults.style.display = 'block';
+      this.searchResults.style.width = facetWidth - 40 + 'px';
+
+      if ($$(this.searchResults).css('display') == 'none') {
+        this.searchResults.style.display = '';
```

Comment:
What does setting the `display` to an empty string mean? Is it to set it to it's default value? If so, would `inherit` work?

Same for line 112.

Reply:
It will remove the style cancel the style applied by the style attribute, so if a stylesheet uses the class, that one will be applied instead.